src/index.js

```diff
@@ -37,16 +40,21 @@ class Routes {
     return {route, params}
   }
 
-  getRequestHandler (app) {
+  getRequestHandler (app, options = {}) {
     const nextHandler = app.getRequestHandler()
+    this.options = Object.assign({}, this.options, options)
 
     return (req, res) => {
       const parsedUrl = parse(req.url, true)
       const {pathname, query} = parsedUrl
       const {route, params} = this.match(pathname)
 
       if (route) {
-        app.render(req, res, route.page, {...query, ...params})
+        if (this.options.routeFound && typeof this.options.routeFound === 'function') {
```

Comment:
What do you think of:

```javascript
if (route) {
  Object.assign(query, params)

  if (handler) {
    handler({req, res, route, query})
  } else {
    app.render(req, res, route.page, query)
  }
}
```
The handler gets an object as single parameter including the full route object